src/main.rs

```diff
@@ -141,7 +150,56 @@ fn check_payload_detections(
     None
 }
 
-async fn handle_generation(
+async fn handle_chat_completions(
+    headers: HeaderMap,
+    Json(payload): Json<serde_json::Value>,
+    detectors: Vec<String>,
+    gateway_config: GatewayConfig,
+    route_fallback_message: Option<String>,
+    orchestrator_client: Arc<reqwest::Client>,
+    scheme: String,
+) -> Result<Response, (StatusCode, String)> {
+    tracing::debug!("handle_chat_completions called with payload: {:?}", payload);
+
+    // Check if streaming is requested
+    let is_streaming = payload
+        .as_object()
+        .and_then(|obj| obj.get("stream"))
+        .and_then(|v| v.as_bool())
+        .unwrap_or(false);
+
+    if is_streaming {
+        let result = handle_streaming_generation(
+            headers,
+            Json(payload),
+            detectors,
+            gateway_config,
+            route_fallback_message,
+            orchestrator_client,
+            scheme,
+        ).await;
+        match result {
+            Ok(response) => Ok(response.into_response()),
+            Err(e) => Err(e),
+        }
+    } else {
+        let result = handle_non_streaming_generation(
+            headers,
+            Json(payload),
+            detectors,
+            gateway_config,
+            route_fallback_message,
+            orchestrator_client,
+            scheme,
+        ).await;
+        match result {
+            Ok(response) => Ok(response.into_response()),
+            Err(e) => Err(e),
+        }
+    }
```

Comment:
The two branches are mostly the same, maybe you could try something like:
```suggestion
    let handler = if is_streaming {
        handle_streaming_generation
    } else {
        handle_non_streaming_generation
    };
    
    let result = handler(
        headers,
        Json(payload),
        detectors,
        gateway_config,
        route_fallback_message,
        orchestrator_client,
        scheme,
    )
    .await;
    
    match result {
        Ok(response) => Ok(response.into_response()),
        Err(e) => Err(e),
    }
}
```